### PR TITLE
Offchain tests

### PR DESCRIPTION
**Pull request by lamafab**
Work-in-Progress. Some functions are left out.

### CODE REVIEW

**Comment by FlorianFranzen** [comment]
Looks like we are running into some problem with gossamer not providing the following functions: 
- `ext_offchain_http_request_write_body_version_1`
- `ext_offchain_http_response_wait_version_1`
- `ext_offchain_http_response_headers_version_1`
- `ext_offchain_http_response_read_body_version_1`

**Review by FlorianFranzen — CHANGES_REQUESTED** [review]
Looks like we are running into some problem with gossamer not providing the following functions: 
- `ext_offchain_http_request_write_body_version_1`
- `ext_offchain_http_response_wait_version_1`
- `ext_offchain_http_response_headers_version_1`
- `ext_offchain_http_response_read_body_version_1`

So since they do not have any tests for them yet, I would just comment them out for now. This will be strictly necessary to merge this, as this will otherwise fail our CI every time.

Secondly, I am not sure how useful these tests are right now, apart from testing encoding. So I think in a next step, we should see what the other implementations return.

File: runtimes/hostapi/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ parity-scale-codec = { version = "3.0", default-features = false, optional = tru
 substrate-wasm-builder = "3.0"
 
 [features]
-std = [ "sp-core/std", "parity-scale-codec/std" ]
+std = [ "sp-core/std", "parity-scale-codec" ]
```
Comment (FlorianFranzen) [inline runtimes/hostapi/Cargo.toml:15]: What is this about?

**Comment by lamafab** [comment]
I should still create some fixtures (where necessary) and integrate it with the Julia tests, on my TODO.

File: adapters/substrate/src/host_api/offchain.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
+use crate::host_api::utils::{str, ParsedInput, Runtime};
+use parity_scale_codec::{Encode, Decode};
+
+#[derive(Debug, Decode)]
+struct OpaqueNetworkState {
+	_peer_id: Vec<u8>,
+	_addresses: Vec<Vec<u8>>,
+}
```
[the remark this answers is not in the document]
Comment (lamafab) [reply adapters/substrate/src/host_api/offchain.rs:8]: It is, but simple enough to replicate and avoid an import. Anyway, currently it just gives us `Err(())` anyway.

**Pull request merged**